src/odemis/acq/test/flim_test.py

```diff
@@ -0,0 +1,205 @@
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)s:%(lineno)d %(message)s")
+
+CONFIG_PATH = os.path.dirname(odemis.__file__) + "/../../install/linux/usr/share/odemis/"
+SECOM_FLIM_CONFIG = CONFIG_PATH + "sim/secom-sim-flim.odm.yaml"
+
+'''
+To run these tests, you must run the Symphotime simulator.
+
+The simulator can be found at https://github.com/PicoQuant/SymPhoTime_Remote_Interface
+The PQ_ServerSimulator application is a Win32 or x64 based exe that simulates a Symphotime server
+By default, it listens on 127.0.0.1:6000 for a connection. Just launch the EXE with default settings
+in order to run these tests.
+'''
+
+
+class TestFlimtatic(unittest.TestCase):
+    backend_was_running = False
+
+    @classmethod
+    def setUpClass(cls):
+
+        try:
+            test.start_backend(SECOM_FLIM_CONFIG)
+        except LookupError:
+            logging.info("A running backend is already found, skipping tests")
+            cls.backend_was_running = True
+            return
+        except IOError as exp:
+            logging.error(str(exp))
+            raise
+
+        # Find CCD & SEM components
+        cls.sft = model.getComponent(role="time-correlator")
+        cls.tc_scanner = model.getComponent(role="tc-scanner")
+        cls.ex_light = model.getComponent(role="light")
+        cls.lscanner = model.getComponent(role="laser-mirror")
+        cls.apd = model.getComponent(role="tc-detector")
+        cls.detector = model.getComponent(role="photo-detector0")
+
+        cls.ex_light.period.value = 0.0001
+        em = [0] * len(cls.ex_light.emissions.value)
+        cls.ex_light.emissions.value = em
+        cls.ex_light.power.value = 0.25
+
+    @classmethod
+    def tearDownClass(cls):
+        time.sleep(1.0)
+
+        if cls.backend_was_running:
+            return
+        test.stop_backend()
+
+    def _on_image(self, im):
+        # Is called when an image is received in the live setting stream
+        logging.info("image: dim %s" , im.shape)
+        self._image = im
+
+    def validate_scan_sim(self, remote, repetition):
+        self.assertIsNotNone(remote.raw)
+        self.assertIsNotNone(remote.raw[0].metadata)
+        
+    def validate_image(self, img):
+        self.assertIsNotNone(img)
+        self.assertIsNotNone(img.metadata)
+
+    def test_flim_acq_simple(self):
+        logging.debug("Testing acquisition")
+
+        helper = stream.ScannedTCSettingsStream('Stream', self.detector, self.ex_light, self.lscanner,
+                                                self.sft, self.apd, self.tc_scanner)
+
+        remote = stream.ScannedRemoteTCStream("remote", helper)
+
+        # Configure values and test acquisition for several dwell times.
+        for dwellTime in (1e-6, 6e-3):
+            helper.dwellTime.value = dwellTime  # seconds
+            helper.repetition.value = (64, 64)
+            helper.roi.value = (0, 0, 0.1, 0.1)
+            helper.image.subscribe(self.validate_image)
```

Comment:
If a subscriber, eg `self.validate_image`, fails, the test will still pass fine (because it's in a separate thread). That's why you preferably want to just store the image in an attribute and validate it in the main thread of the unit test.